MIGRATION_GUIDE.md

```diff
@@ -1,162 +1,21 @@
-- `SetWebhook::url`: `String` -> `Url`
-
-List of changed optional params:
-- `AnswerCallbackQuery::url`: `String` -> `Url`
-- `SendInvoice::photo_url`: `String` -> `Url`
-- `CreateChatInviteLink::expire_date`: `i64` -> `DateTime<Utc>` 
-- `EditChatInviteLink::expire_date`: `i64` -> `DateTime<Utc>` 
-- `KickChatMember::until_date`: `u64` -> `DateTime<Utc>` 
-- `RestrictChatMember::until_date`: `u64` -> `DateTime<Utc>` 
-- `SendPoll::close_date`: `u64` -> `DateTime<Utc>` 
-
-
-#### Renamed items
-
-Some items (fields, variants, types, methods) were renamed.
-If you used them, you should start using new names.
-
-Example:
-```diff
--bot.send_chat_action(chat, ChatAction::RecordAudio).await?;
-+bot.send_chat_action(chat, ChatAction::RecordVoice).await?;
-
--if chat_member.is_kicked() {
-+if chat_member.is_banned() {
-    /* ... */
-}
-```
-
-List of renamed items:
-- `ChatAction::RecordAudio` -> `RecordVoice`
-- `ChatAction::UploadAudio` -> `UploadVoice`
-- `ChatMemberKind::Creator` -> `Owner`
-- `ChatMemberKind::Kicked` -> `Banned`
-- `Creator` -> `Owner`
-- `Kicked` -> `Banned`
-- `ChatMemberKind::is_Creator` -> `is_owner` *
-- `ChatMemberKind::is_kicked` -> `is_banned` *
-- `ChatMemberStatus::Creator` -> `Owner`
-- `ChatMemberStatus::Kicked` -> `Banned`
-- `kick_chat_member` -> `ban_chat_member` *
-- `get_chat_members_count` -> `get_chat_member_count` *
-
-\* Old methods are still accessible, but deprecated
-
-
-#### Added `impl Clone` for {`CacheMe`, `DefaultParseMode`, `Throttle`}
-
-Previously said bot adaptors were lacking `Clone` implementation. 
-To workaround this issue it was proposed to wrap bot in `Arc`.
-Now it's not required, so you can remove the `Arc`:
-
-```diff
-let bot = Bot::new(token).parse_mode(ParseMode::MarkdownV2);
--let bot = Arc::new(bot);
-```
-
-
-### teloxide
-
-#### Mutable reference for dispatching
-
-`Dispatcher::dispatch` and `Dispatcher::dispatch_with_listener` now require mutable (unique) reference to self.
-If you've used variable to store `Dispatcher`, you need to make it mutable:
-
-```diff
--let dp = Dispatcher::new();
-+let mut dp = Dispatcher::new();
-/* ... */
-dp.dispatch();
-```
-
-
-#### Listener refactor
-
-`UpdateListener` trait was refactored.
-If you've used `polling`/`polling_default` provided by teloxide, no changes are required.
-If, however, you've used or implemented `UpdateListener` directly or used a `Stream` as a listener, 
-then you need to refactor your code too.
-
-See also: [teloxide examples fixes](https://github.com/teloxide/teloxide/pull/385/files/8785b8263cb4caebf212e2a66a19f73e653eb060..c378d6ef4e524da96718beec6f989e8ac51d1531).
-
-
-#### `polling_default`
-
-`polling_default` is now async, but removes webhook.
-
-Example fix:
-```diff
--let listener = polling_default(bot);
-+let listener = polling_default(bot).await;
-```
+[chain of responsibility pattern]: https://en.wikipedia.org/wiki/Chain-of-responsibility_pattern
+[`dptree`]: https://github.com/p0lunin/dptree
```

Comment:
Please, adds notes about BCs from core:
- `InputFile` now can't be created like `InputFile::Url(url)` or matched on, use constructors like `InputFile::url`, `InputFile::file`, etc
- `RequestError` and `DownloadError` error variants were slightly renamed
- `ChatPermissions` is now bitflags

Reply:
Fixed.